### PR TITLE
Configure Spring MVC, Create UserResponseDto, Implement UserController

### PR DESCRIPTION
**Pull request by IrynaDim**
I left the jsp page because it is more convenient to click on the "Inject data into the DB" or "Get all users" button than to write path in the address line.

### CODE REVIEW

**Review by RomanHreits — APPROVED** [review]

File: src/main/java/spring/intro/config/WebConfig.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.web.servlet.view.InternalResourceViewResolver;
+import org.springframework.web.servlet.view.JstlView;
+
+@Configuration
+@EnableWebMvc
+@ComponentScan(basePackages = {"spring.intro.controllers"})
+public class WebConfig {
+
+    @Bean
+    public InternalResourceViewResolver resolver() {
```
Comment (Sofasmile) [inline src/main/java/spring/intro/config/WebConfig.java:16]: we can remove this method

File: src/main/java/spring/intro/controllers/MainPageController.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package spring.intro.controllers;
+
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+
+@Controller
+public class MainPageController {
```
Comment (Sofasmile) [inline src/main/java/spring/intro/controllers/MainPageController.java:7]: and this controller can be removed too

File: src/main/java/spring/intro/controllers/user/UserController.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+import org.springframework.web.bind.annotation.RestController;
+import spring.intro.config.AppConfig;
+import spring.intro.dto.UserResponseDto;
+import spring.intro.model.User;
+import spring.intro.service.UserService;
+
+@RestController
+@RequestMapping("/user")
+public class UserController {
+    AnnotationConfigApplicationContext context =
```
Comment (Sofasmile) [inline src/main/java/spring/intro/controllers/user/UserController.java:18]: let's use constructor injection

File: src/main/java/spring/intro/controllers/user/UserController.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    @GetMapping
+    public List<UserResponseDto> getAll() {
+        List<User> allUsers = userService.listUsers();
+        List<UserResponseDto> allUsersDto = allUsers.stream()
+                .map(user -> changeToResponseDto(user))
+                .collect(Collectors.toList());
+        return allUsersDto;
+    }
+
+    private UserResponseDto changeToResponseDto(User user) {
```
Comment (Sofasmile) [inline src/main/java/spring/intro/controllers/user/UserController.java:46]:
```suggestion
    private UserResponseDto convertToResponseDto(User user) {
```

File: src/main/java/spring/intro/controllers/user/UserController.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        return changeToResponseDto(user);
+    }
+
+    @GetMapping
+    public List<UserResponseDto> getAll() {
+        List<User> allUsers = userService.listUsers();
+        List<UserResponseDto> allUsersDto = allUsers.stream()
+                .map(user -> changeToResponseDto(user))
+                .collect(Collectors.toList());
+        return allUsersDto;
```
Comment (Sofasmile) [inline src/main/java/spring/intro/controllers/user/UserController.java:43]:
```suggestion
        return userService.listUsers().stream()
                .map(user -> changeToResponseDto(user))
                .collect(Collectors.toList());
```

File: src/main/webapp/WEB-INF/views/MainPage.jsp [hunk 6]
```diff
@@ -0,0 +1,14 @@
+<%@ page contentType="text/html;charset=UTF-8" language="java" %>
```
Comment (Sofasmile) [inline src/main/webapp/WEB-INF/views/MainPage.jsp:1]: let's remove jsp page

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**